app/controllers/reactions_controller.rb

```diff
@@ -102,11 +102,13 @@ def create
     render json: { result: result, category: category }
   end
 
-  def cached_user_public_reactions(user, comment_ids)
-    cache = Rails.cache.fetch("cached-user-reactions-#{user.id}-#{user.public_reactions_count}", expires_in: 24.hours) do
-      JSON.parse(user.reactions.public_category.to_json)
+  def cached_user_public_comment_reactions(user, comment_ids)
+    cache = Rails.cache.fetch("cached-user-#{user.id}-reaction-ids-#{user.public_reactions_count}", expires_in: 24.hours) do
+      user.reactions.public_category.each_with_object({}) do |r, h|
+        h[r.reactable_id] = r.attributes
+      end
     end
-    cache.select { |a| comment_ids.include?(a["reactable_id"]) }
+    cache.slice(*comment_ids).values.select { |r| r["reactable_type"] == "Comment" }
```

Comment:
I would add this to your where statement. 
```
user.reactions.public_category.where(reactable_type: "Comment")
```

Reply:
Yep I did. I think you were GitHub was showing you my older commit 😄